packages/types/src/feature-ids.ts

```diff
@@ -0,0 +1,18 @@
+/**
+ * @internal
+ */
+export type SmithyFeatures = Partial<{
+  RESOURCE_MODEL: "A";
+  WAITER: "B";
+  PAGINATOR: "C";
+  RETRY_MODE_LEGACY: "D";
+  RETRY_MODE_STANDARD: "E";
+  RETRY_MODE_ADAPTIVE: "F";
+  GZIP_REQUEST_COMPRESSION: "L";
+  PROTOCOL_RPC_V2_CBOR: "M";
+  ENDPOINT_OVERRIDE: "N";
+  SIGV4A_SIGNING: "S";
+  CREDENTIALS_CODE: "e";
+  CREDENTIALS_HTTP: "z";
+  CREDENTIALS_IMDS: "0";
```

Comment:
some of these credentials features aren't technically Smithy, but they live in this repository as of the migration from AWS SDK.

Reply:
The full list can be seen in the paired PR https://github.com/aws/aws-sdk-js-v3/pull/6536/files

under a similar file `types/src/feature-ids.ts`

Reply:
> as of the migration from AWS SDK

I'm not sure what that means. If there's a separate list over there, why do AWSisms need to live here?

Reply:
All the JS code used to be in AWS SDK. The generic or core runtime packages were moved over to smithy but it wasn't completely separated and the instance metadata provider exists in the smithy-typescript repo. So, that feature may need to be known by the Smithy list of features.

However, I think I can remove credentials imds and credentials http feature detection from smithy and do it in the SDK code. I've pushed a commit to remove these two.